ffi/capi/src/time_formatter.rs

```diff
@@ -51,9 +51,7 @@ pub mod ffi {
         #[diplomat::rust_link(icu::datetime::fieldsets::T::short, FnInStruct, hidden)]
         #[diplomat::rust_link(icu::datetime::fieldsets::T::medium, FnInStruct, hidden)]
         #[diplomat::rust_link(icu::datetime::fieldsets::T::long, FnInStruct, hidden)]
-        #[diplomat::rust_link(icu::datetime::fieldsets::T::time, FnInStruct, hidden)]
-        #[diplomat::rust_link(icu::datetime::fieldsets::T::time_hm, FnInStruct, hidden)]
-        #[diplomat::rust_link(icu::datetime::fieldsets::T::time_hms, FnInStruct, hidden)]
+        #[diplomat::rust_link(icu::datetime::fieldsets::T::with_time_precision, FnInStruct, hidden)]
```

Comment:
@sffc there's an inconsistency here, `fieldsets::T` doesn't have `time`, `time_hm`, `time_hms`, but the codegen wants it to be consistent with other fieldsets. Is this intended?

Reply:
Instead of `T::time`, `T::time_hm`, and `T::time_hms` it has `T::hm` and `T::hms` (and maybe should have `T::for_time_precision`). So I think codegen should not include the ones it doesn't have.

Reply:
If you make a list of the dead links in the datetime modules, I'm wiling to fix them in a follow up.

How did you generate the dead links? We should probably automate and enforce that.

Reply:
It's just those three, plus `NoCalendarFormatter::try_new` needs to be `FnInTypedef`, not `FnInStruct`.

I did the set difference in the other direction, but that will also find all links that point to allowlist types, so we'd have to change the docs parsing which is too complicated now.